src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
     {
         return;
     }
-            
+
+    bool bindBufferMemory(uint32_t bindInfoCount, const SBindBufferMemoryInfo* pBindInfos) override
+    {
+        for (uint32_t i = 0u; i < bindInfoCount; ++i)
+        {
+            if ((pBindInfos[i].buffer->getAPIType() != EAT_VULKAN) /*|| (pBindInfos[i].memory->getAPIType() != EAT_VULKAN)*/)
+                continue;
+
+            VkBuffer vk_buffer = static_cast<const CVulkanBuffer*>(pBindInfos[i].buffer)->getInternalObject();
+            VkDeviceMemory vk_memory = static_cast<const CVulkanMemoryAllocation*>(pBindInfos[i].memory)->getInternalObject();
+            if (vkBindBufferMemory(m_vkdev, vk_buffer, vk_memory, static_cast<VkDeviceSize>(pBindInfos[i].offset)) != VK_SUCCESS)
+                return false;
```

Comment:
dont quit on first fail

Also you need to update the state of `CVulkanBuffer` (make it hold the smart pointer to the bound memory)

Reply:
resolved

Reply:
@Erfan-Ahmadi its still wrong, the `vulkanBuffer->setMemoryAndOffset` should only be called iff the `vkBindBufferMemory` call succeeds